mitre-attack/utils.py

```diff
@@ -42,6 +42,26 @@ def query_source(memory_source, query_filter):
     return result
 
 
+def filter_techniques(techniques_all):
+    # Sometimes techniques do not have the 'x_mitre_is_subtechnique' key that determines if they are subtechniques or not
+    # These are actually still techniques but get dropped by our filters.
+    # This function finds such techniques adds them back into our result
+    techniques_without_sub_key = []
+    techniques_with_sub_key = []
+    sub_techniques = []
+
+    for technique in techniques_all:
+        if 'x_mitre_is_subtechnique' not in technique:
+            techniques_without_sub_key.append(technique)
+        else:
+            if technique['x_mitre_is_subtechnique']:
+                sub_techniques.append(technique)
+            else:
+                techniques_with_sub_key.append(technique)
+
+    return techniques_without_sub_key + techniques_with_sub_key, sub_techniques
```

Comment:
the join creates a third array, increasing the memory usage further since the list of objects is already high. why not put both into the techniques list while iterating itself?